[CODE] declassify game

rustykitty opened the pull request:
<!-- Write BELOW The Headers and ABOVE The comments else it may not be viewable. -->
<!-- You can view CONTRIBUTING.md for a detailed description of the pull request process. -->
<!-- Be sure to name your PR something descriptive and succinct; include Bugfix: Feature: Enhancement: or Content: in the title to describe what type of PR it is. -->

## About The Pull Request
Moved `game` to its own file and deleted the `game_essentials` file and updated usages accordingly. Did some code cleanup and formatting too.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage senior developers from merging your PR! -->

## Why This Is Good For ClanGen

<!-- If this is a bug fix, you can remove this section. -->
<!-- Please add a short description of why you think these changes would benefit the game. -->
<!-- If this PR was the result of discussion/testing on the discord, please add a link to the discord conversation here. -->
<!-- If you have multiple features that can stand on their own, or unrelated bugfixes, please create separate PRs for them. -->

## Linked Issues
#3412
<!-- If this is unrelated to an issue, you can remove this section. -->
<!-- If this was in response to a GitHub issue, please write it here with the format Fixes: #1234 so that GitHub knows to link the issues. -->
<!-- If this PR was the result of discussion/testing on the discord, please add a link to the discord conversation here. -->

## Proof of Testing
Game seems to run fine.
<img width="912" alt="Screenshot 2025-02-07 at 9 11 21 PM" src="https://github.com/user-attachments/assets/521a6d4c-ad2d-4bb4-9898-ceb94ff42ee4" />

<img width="912" alt="Screenshot 2025-02-07 at 9 11 41 PM" src="https://github.com/user-attachments/assets/98fca903-6629-4d11-8abd-71d916d2dec1" />

<img width="912" alt="Screenshot 2025-02-07 at 9 12 23 PM" src="https://github.com/user-attachments/assets/6534e1b7-155a-4190-9b90-e76216c26134" />

<!-- Include any screenshots, debugging steps, or links to beta testing threads here. At least one form of proof of testing is REQUIRED for all new content. You must be able to run the code locally before you PR it here. -->

## Changelog/Credits

<!-- Include any changes that should be made to the changelog of the game here or any changes to the credits file of the game. -->
<!-- This is just for easy access later for senior developers gathering this information; this process is not automated. -->

rustykitty committed:
quick and dirty solution?

rustykitty committed:
missed some

rustykitty committed:
scope

rustykitty committed:
newlines between methods

rustykitty committed:
fix import

rustykitty committed:
slightly clearer code

j-gynn commented:
Looking good so far! Because this is a somewhat significant change to the codebase, we'd like for you to demonstrate some more thorough testing than just "game seems to work" 😉 I feel instinctively that it *shouldn't* cause problems, but "shouldn't" and "doesn't" are two very different beasts, especially in code...!

Please create a thread in the Discord for beta testing (reach out to me or any other non-hiatused senior dev if you want help doing this). Once you have, please link it here under the testing section. I know this is only a draft PR so if you're not completely ready for beta-testing just yet that's totally fine!

rustykitty committed:
fix import?

rustykitty committed:
trying again

rustykitty committed:
Merge remote-tracking branch 'upstream/development' into declassify-game

rustykitty committed:
Merge branch 'development' into declassify-game

rustykitty commented:
Proof of testing: https://discord.com/channels/1003759225522110524/1054942209260146818/1358278730694397963

rustykitty committed:
Merge remote-tracking branch 'upstream/development' into declassify-game

rustykitty committed:
Merge remote-tracking branch 'upstream/development' into declassify-game

# Conflicts:
#	scripts/cat/cats.py
#	scripts/cat/history.py
#	scripts/cat/names.py
#	scripts/cat/pelts.py
#	scripts/cat/sprites.py
#	scripts/clan.py
#	scripts/clan_resources/freshkill.py
#	scripts/clan_resources/herb/herb_supply.py
#	scripts/debug_commands/cat_pregnancy.py
#	scripts/debug_commands/fps.py
#	scripts/debug_commands/settings.py
#	scripts/events.py
#	scripts/events_module/event_filters.py
#	scripts/events_module/generate_events.py
#	scripts/events_module/ongoing/disaster_events.py
#	scripts/events_module/patrol/patrol.py
#	scripts/events_module/relationship/pregnancy_events.py
#	scripts/events_module/relationship/relation_events.py
#	scripts/events_module/relationship/romantic_events.py
#	scripts/events_module/short/condition_events.py
#	scripts/events_module/short/handle_short_events.py
#	scripts/game_structure/audio.py
#	scripts/game_structure/discord_rpc.py
#	scripts/game_structure/game_essentials.py
#	scripts/game_structure/load_cat.py
#	scripts/game_structure/localization.py
#	scripts/game_structure/screen_settings.py
#	scripts/game_structure/ui_elements.py
#	scripts/game_structure/windows.py
#	scripts/screens/CeremonyScreen.py
#	scripts/screens/ChangeGenderScreen.py
#	scripts/screens/ChooseAdoptiveParentScreen.py
#	scripts/screens/ChooseMateScreen.py
#	scripts/screens/ChooseMentorScreen.py
#	scripts/screens/ClanScreen.py
#	scripts/screens/ClanSettingsScreen.py
#	scripts/screens/EventsScreen.py
#	scripts/screens/FamilyTreeScreen.py
#	scripts/screens/LeaderDenScreen.py
#	scripts/screens/ListScreen.py
#	scripts/screens/MakeClanScreen.py
#	scripts/screens/PatrolScreen.py
#	scripts/screens/RelationshipScreen.py
#	scripts/screens/Screens.py
#	scripts/screens/SettingsScreen.py
#	scripts/screens/StartScreen.py
#	scripts/screens/WarriorDenScreen.py
#	scripts/screens/screens_core/screens_core.py
#	scripts/special_dates.py
#	tests/test_cat.py
#	tests/test_save.py

rustykitty committed:
Fix imports, redo declassify based on latest changes

moved game_structure.game_essentials to game_structure.game.__init__

rustykitty committed:
fix event_filter import

scribblecrumb commented on scripts/game_structure/game/__init__.py:
```diff
@@ -0,0 +1,286 @@
+patrol_cats = {}
+patrolled = []
+
+outsider_reps = ["welcoming", "neutral", "hostile"]
+other_clan_reps = ["ally", "neutral", "hostile"]
+
+BIOME_TYPES = ["Forest", "Plains", "Mountainous", "Beach", "Wetlands", "Desert"]
+
```
We should be able to delete these (`outsider_reps`, `other_clan_reps`, `BIOME_TYPES`), since we have them stored in `constants.py` now

scribblecrumb commented on scripts/game_structure/game/__init__.py:
```diff
@@ -0,0 +1,286 @@
+# max_allegiance_displayed = 17
+# allegiance_scroll_ct = 0
+# max_relation_events_displayed = 10
+# relation_scroll_ct = 0
```
Let's just remove these, atp I don't think there's any reason to save them

scribblecrumb commented on scripts/game_structure/game/__init__.py:
```diff
@@ -0,0 +1,286 @@
+
+    try:
+        with open(
+            get_save_dir() + "/" + clan.name + "/faded_cats/" + parent + ".json",
```
Could we convert this to an f-string?

scribblecrumb reviewed (changes requested):
Thank you for being so diligent in keeping this up to date! Sorry it took so long to get around to.  Just a few suggestions from me and then resolving the remain conflicts and I think we're good to go on this.

rustykitty committed:
apply suggested changes to file

Also, made path a variable in `add_faded_offspring_to_faded_cat` because it's used twice

rustykitty committed:
Merge remote-tracking branch 'upstream/development' into declassify-game

# Conflicts:
#	main.py
#	scripts/cat_relations/relationship.py
#	scripts/events_module/event_filters.py
#	scripts/events_module/relationship/group_events.py
